### PR TITLE
taktuk: init at 3.7.5

### PR DESCRIPTION
**Pull request by bzizou**
###### Things done:
- [ ] Tested using sandboxing (`nix-build --option build-use-chroot true` or [nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS)
- [X] Built on platform(s): Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).
###### More

Fixes issue #<insert id>

cc @<maintainer>

---

_Please note, that points are not mandatory, but rather desired._

### CODE REVIEW

**Comment by lancelotsix** [comment]
I fail to build it locally

```
…
mkdir -p Perl-Module/lib/
cat DISCLAIMER >Perl-Module/lib/TakTuk.pm
./taktuk -p TakTuk >>Perl-Module/lib/TakTuk.pm
/nix/store/pnc8xc1dh3cy4migfipfmzy3ad7snz99-bash-4.3-p42/bin/bash: ./taktuk: /usr/bin/perl: bad interpreter: No such file or directory
Makefile:1009: recipe for target 'Perl-Module/lib/TakTuk.pm' failed
make[1]: *** [Perl-Module/lib/TakTuk.pm] Error 126
make[1]: Leaving directory '/tmp/nix-build-taktuk-3.7.5.drv-0/taktuk-3.7.5'
Makefile:509: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1
builder for ‘/nix/store/g5gbljdgj2vwvb23ihab8fy1xzghqxc7-taktuk-3.7.5.drv’ failed with exit code 2
error: build of ‘/nix/store/g5gbljdgj2vwvb23ihab8fy1xzghqxc7-taktuk-3.7.5.drv’ failed
```

Maybe you should add `perl` to the `buildInputs` and add something `substituteInPlace taktuk "/usr/bin/perl" "${perl}/bin/perl` in the `patchPhase` ?

File: pkgs/applications/networking/cluster/taktuk/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{ stdenv, fetchurl, perl , openssh}:
+
+stdenv.mkDerivation rec {
+  version = "3.7.5";
+  name = "taktuk-${version}";
+
+  buildInputs = [ perl ];
+
+  src = fetchurl {
+    url = "https://gforge.inria.fr/frs/download.php/33412/taktuk-3.7.5.tar.gz";
```
Comment (lancelotsix) [inline pkgs/applications/networking/cluster/taktuk/default.nix]: `${name}` could be used here to reduce duplications.

Comment (bzizou) [inline pkgs/applications/networking/cluster/taktuk/default.nix]: OK, for ${name}.
But for the build, it's weird:
- perl is already in the buildInputs
- It builds ok for me, and the resulting perl script uses the good perl binary:
```
    bzizou@killeen:~/nixpkgs$ head -1 $(which taktuk)                      
    #!/nix/store/s5hmy8kv1w151a1z4wy57csla3c11jr9-perl-5.20.3/bin/perl -w 
```
And during the build, I can see:
`checking for perl... /nix/store/s5hmy8kv1w151a1z4wy57csla3c11jr9-perl-5.20.3/bin/perl`

I can patch, but why in your case does the build uses /usr/bin/perl?

**Comment by bzizou** [comment]
it's weird:

perl is already in the buildInputs
It builds ok for me, and the resulting perl script uses the good perl binary:
    bzizou@killeen:~/nixpkgs$ head -1 $(which taktuk)                      
    #!/nix/store/s5hmy8kv1w151a1z4wy57csla3c11jr9-perl-5.20.3/bin/perl -w 
And during the build, I can see:
checking for perl... /nix/store/s5hmy8kv1w151a1z4wy57csla3c11jr9-perl-5.20.3/bin/perl

I can patch, but why in your case does the build uses /usr/bin/perl?

**Comment by lancelotsix** [comment]
OK for the first one:

```
checking for perl... /nix/store/s5hmy8kv1w151a1z4wy57csla3c11jr9-perl-5.20.3/bin/perl
```

but apparently, the Makefile calls the `./taktuk` script which have `/usr/bin/perl` as sheebang which is unavailalble.

To have the build pass I have to add something like

```
  preBuild = ''
    substituteInPlace ./taktuk --replace "/usr/bin/perl" "${perl}/bin/perl"
  '';
```

(I only tested the build with that, not the results)

**Comment by bzizou** [comment]
OK pour your preBuild. I added it and still builds for me, and produces a working result. So, amended.

**Comment by lancelotsix** [comment]
Merged, and I also changed the URL for `https://gforge.inria.fr/frs/download.php/33412/${name}.tar.gz` (even it it strangely accepts to fetch "taktuk-3.7.5-3.7.5.tar.gz")

**Comment by bzizou** [comment]
> (even it it strangely accepts to fetch "taktuk-3.7.5-3.7.5.tar.gz")

oops :-) Thanx!

**Pull request merged**